packages/@aws-cdk/aws-cognito/lib/user-pool-idp.ts

```diff
@@ -0,0 +1,31 @@
+import { Construct, IResource, Resource } from '@aws-cdk/core';
+
+/**
+ * Represents a UserPoolIdentityProvider
+ */
+export interface IUserPoolIdentityProvider extends IResource {
+  /**
+   * The primary identifier of this identity provider
+   * @attribute
+   */
+  readonly providerName: string;
```

Comment:
This could be typed `UserPoolClientIdentityProvider`.

Reply:
I'm not so confident that they are one and the same. They are the same at the moment, for the providers I've chosen to implement. 

Ex: the list [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cognito-userpoolclient.html#cfn-cognito-userpoolclient-supportedidentityproviders) does not match up with [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cognito-userpoolclient.html#cfn-cognito-userpoolclient-supportedidentityproviders).

It's not clear yet if this is a documentation gap, or something more.

Leaving this as is for now.